Solutions/Argotic.Extensions/Core/AtomPublishing/AtomPublishingControlSyndicationExtension.cs

```diff
@@ -89,9 +89,14 @@ public static bool MatchByType(ISyndicationExtension extension)
 		/// <exception cref="ArgumentNullException">The <paramref name="source"/> is a null reference (Nothing in Visual Basic).</exception>
 		public override bool Load(IXPathNavigable source)
 		{
+			bool wasLoaded = false;
```

Comment:
You've absolutely managed to identify where the problem is, but there is an easier way to fix this!

Reply:
Rather than making these changes, if you go to `Load(IXPathNavigable source, SyndicationResourceLoadSettings settings)` you can change the method to:

```C#
public bool Load(IXPathNavigable source, SyndicationResourceLoadSettings settings)
{
    Guard.ArgumentNotNull(source, "source");

    XPathNavigator navigator = source.CreateNavigator();
    bool wasLoaded = Context.Load(navigator, CreateNamespaceManager(navigator));
    SyndicationExtensionLoadedEventArgs args = new SyndicationExtensionLoadedEventArgs(source, this);
    this.OnExtensionLoaded(args);

    return wasLoaded;
}
```

The problem is that this method initialised the settings = which defaults to autoloading the extensions, which leads to us getting caught in a recursive loop. We need to remove the settings and use the overload that doesn't use settings.

Reply:
Actually take 3... in the above we're now ignoring settings, but instead we should not make the assumption of creating default settings... rather detect if settings are null and then call the overload that doesn't take settings:

```C#
public bool Load(IXPathNavigable source, SyndicationResourceLoadSettings settings)
{
    Guard.ArgumentNotNull(source, "source");
    XPathNavigator navigator = source.CreateNavigator(); ;

    bool wasLoaded;

    if (settings != null)
    {
        wasLoaded = Context.Load(navigator, CreateNamespaceManager(navigator), settings);
    }
    else
    {
        wasLoaded = Context.Load(navigator, CreateNamespaceManager(navigator));
    }

    SyndicationExtensionLoadedEventArgs args = new SyndicationExtensionLoadedEventArgs(source, this);
    this.OnExtensionLoaded(args);

    return wasLoaded;
}
```